csharp-basic-topics/ForVsForeachInCSharp/ForVsForeachInCSharp/ForVsForeachExamples.cs

```diff
@@ -0,0 +1,126 @@
+﻿using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.Order;
+
+namespace ForVsForeachInCSharp;
+
+[MemoryDiagnoser]
+[RankColumn]
+[Orderer(SummaryOrderPolicy.FastestToSlowest)]
+[GroupBenchmarksBy(BenchmarkLogicalGroupRule.ByParams)]
+
+public class ForVsForeachExamples
+{
+    private readonly GenerateData _generateData = new();
+    private readonly int[] _intArray;
+    private readonly List<int> _intList;
+    private readonly Stack<int> _intStack;
+    private readonly Queue<int> _intQueue;
+    private readonly Dictionary<int, int> _intDictionary;
+
+    public ForVsForeachExamples()
+    {
+        _intArray = _generateData.GenerateRandomArray(100000);
+        _intList = _generateData.GenerateRandomList(100000);
+        _intStack = _generateData.GenerateRandomStack(100000);
+        _intQueue = _generateData.GenerateRandomQueue(100000);
+        _intDictionary = _generateData.GenerateRandomDictionary(100000);
+    }
+
+    [Benchmark]
+    public void PrintArrayUsingForeach()
+    {
+        foreach (var element in _intArray)
+        {
+            Console.WriteLine(element);
+        }
+    }
+
+    [Benchmark]
+    public void PrintArrayUsingForLoop()
+    {
+        for (int i = 0; i < _intArray.Length; i++)
+        {
+            Console.WriteLine(_intArray[i]);
+        }
+    }
+
```

Comment:
This applies to all of the benchmark methods. We should not be using Console.WriteLine in a benchmark. We should change the method signature to return an int or a long if that is better and then change the benchmark methods to compute a sum or an average.